### DISCUSSION TITLE
is the ACSI support in code right now?

### DISCUSSION THREAD

**Discussion by dna2496** [MiSTle-Dev/MiSTeryNano#15] [Q&A] [answered]
I compiled the actual git with the GOWIN Designer. All woked fine but whats the way to use the harddisk? How to name the image on the sd? Sorry i dont want to bother you or stress. I am just curious and try to test :) 

Regardes Andreas

**Comment by harbaum** [comment, marked as answer]
Yes, ACSI is now included with the source code. This also has a M0S side and you thus need to compile and update the firmware as well.

Default file names are then acsi0.hd and acsi1.hd but that's less important now as the latest firmware remembers permanently which images were selected.

**Comment by dna2496** [comment]
I flashed both the nano and the bl wth the current version. All work fine but the harddisk dont show up.

**Reply by harbaum** [reply]
How did you name the image file? It needs to end with .hd . Then you can select it from the OSD under drives/acsi #0

**Reply by dna2496** [reply]
theres no drives in the F12 Menu. I named it acsi0.hd. its an Image file from my pistorm Atari. And i flahed the BL twice maybe i do something wrong with the BLDEVCUBE but last time it workes like a charm

**Reply by harbaum** [reply]
Then you are not running the current firmware version.

**Comment by dna2496** [comment]
uh i think i write the bl firmware on the nano itself now it wont show up to flash the fpga itself ... i have to look now how to set it to default. so yes my fault.

**Reply by dna2496** [reply]
"Finally the BL616 MCU is re-flashed with the USB HID firmware. This way you'll loose the ability to flash the FPGA! Before being able to re-flash the FPGA you need to re-install the original firmware which is not publicly available at the moment!" ... means i am fuxxed up?

**Comment by harbaum** [comment]
I got permission to release a binary of the original firmware. You should be able to get it fixed using the "friend 20k":

https://github.com/harbaum/MiSTeryNano/tree/main/bl616/friend_20k

**Reply by dna2496** [reply]
Good Morning, you made my day! Did work .. i was searching the web and leave a comment on the Sipeed website. Thank you a lot i´ll try the Atari later and let you know.

**Reply by dna2496** [reply]
Perfect all looks good. I see the new OSD and can choose the harddisk file but it shows up but wont boot it. I try different things. SD formated at FAT .. works with 2GB limit Fat32 works too exFat doesnt work. I bet i have a wrong image file for the HD file. its an .img and it works on the pistorm Atari emulator. 

If you like i can try to write the windows installation text.

**Reply by harbaum** [reply]
Indeed, exFAT was not enabled. I just enabled it and it seems to work just fine. In the next release exFAT support will be included.

**Reply by dna2496** [reply]
cool, ive tryed all images from the formum none work. i flashed all again and didnt change anything .. cant load from acsi. i try some other TOS versions later.

**Comment by harbaum** [comment]
The "ACSI version in 1 disk of more than 1GB" from here works e.g. out of the box for me:

https://www.atari-forum.com/viewtopic.php?t=34967

And yes, please write a nice installation text. If possible as a markdown which I can include here on github or even as a pull-request.

**Reply by dna2496** [reply]
ok ill try to give my best ;) and thanks for your support

**Reply by dna2496** [reply]
ok done :) in .md but i have no clue how to make a pull-request and the harddisk wont boot ... still trying

**Reply by harbaum** [reply]
Did you try the petera image?

**Reply by harbaum** [reply]
To make a PR you clone my repository into your own GitHub account. Then you do all the changes you want and then you click on "create pull request".

**Reply by dna2496** [reply]
yes i try all of them. maybe i try to fomat the sd card fat again. and ill try the cloene is there another way (discord or something to get in touch)

**Reply by harbaum** [reply]
There's not much to help here. Install  the latest core and firmware, put the hd image onto the card and run it ... 

There's no real way to debug that.

**Comment by harbaum** [comment]
I just checked with ST + TOS 1.04 and STE + TOS 1.62 and both boot the petari game image just fine. I also flashed the core and firmware from the github release to make sure that I didn't upload a broken version.

**Reply by dna2496** [reply]
I reflashed the atarist.fs again with a erase before. now the harddisk works .. with most of the images i have. I started some benchmarks and they worked but the gfx speed seem to be slower than the STE (saved benchmark from my original STE on gembench) and there are gfx glitches when starting demos or games and most crash with some fault code. The diskimage on the sd looks ok but when i open it it have lots of rrrrrr.rrr folders and a size of 623191328 Bytes. I have to look if its the emutos version i tryed before so if it helps i would make a list of things i find while testing

**Comment by harbaum** [comment]
Excellent. I have never tested EmuTOS on this. Fine, if it works. But it will IMHO neither give the same results as a "real" TOS in benchmarks, nor will it run many games or demos. MiSTeryNano is meant to be an exact replica and thus is meant to run under a genuine TOS.

**Reply by dna2496** [reply]
I ll test with the Atari TOS 2.06 on STE and 1.04 on ST

**Reply by harbaum** [reply]
I just noticed that hddriver bombs on MiSTeryNano STE with TOS 2.06. Maybe something is broken under TOS 2.06 I need to fix. You should use 1.62 for STE until then.

**Reply by dna2496** [reply]
k thanks for the info. Did some tests on ST and Mega ST with 1.04 looks good so far. Acsi works and Disk too. Speed on gembench is 100% to my Mega ST. Games run but i didnt connect a Joystick so cant test them but they run. i´ll try TOS 1.62 its the same i have on the original STE i think
